File: 해시_완주하지못한선수/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    string name = "";
    int num = 0;
    int flag = 0;
    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());
    for(int i = 0;i < completion.size(); i++){
        name = participant[i];
        if(participant[i] != completion[i]){
            answer = participant[i];
            flag = 1;
            break;
        }
        
    }
    if (flag ==0)
        answer = participant[participant.size()-1];
    return answer;
}

File: 프로그래머스/해시_완주하지못한선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    data = dict()
    end_data = dict()
    for name in participant:
        data[name] = 0
        end_data[name] = 0
    for name in participant:  
        data[name] +=1
    for name in completion:
        end_data[name]+=1
    for name in end_data:
        if data[name] - end_data[name] != 0:
            answer = name
    return answer